src/forecast.py

```diff
@@ -213,8 +199,112 @@ def forecast_page():
             )
         )
 
-    st.plotly_chart(fig, theme="streamlit")
-
+        start_datetime_utc, end_datetime_utc = determine_start_and_end_datetimes(start_datetimes, end_datetimes)
+
+        if start_datetime_utc and end_datetime_utc:
+            # Initialize Elexon API client
```

Comment:
could we refactor and move to a new file, perhaps in https://github.com/openclimatefix/analysis-dashboard/tree/main/src/plots but call `elexon`, `fetch_forecast_data` could also move there